src/common/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,51 @@
+"use client";
+
+import * as React from "react";
+import * as PopoverPrimitive from "@radix-ui/react-popover";
+import { popoverTheme, type PopoverVariants } from "./Popover.theme";
+
+const PopoveRoot = PopoverPrimitive.Root;
+
+const PopoverTrigger = PopoverPrimitive.Trigger;
+
+const PopoverContent = React.forwardRef<
+  React.ElementRef<typeof PopoverPrimitive.Content>,
+  React.ComponentPropsWithoutRef<typeof PopoverPrimitive.Content>
+>(({ className, align = "center", sideOffset = 4, ...props }, ref) => (
+  <PopoverPrimitive.Portal>
+    <PopoverPrimitive.Content
+      ref={ref}
+      align={align}
+      sideOffset={sideOffset}
+      className={className}
+      {...props}
+    />
+  </PopoverPrimitive.Portal>
+));
+PopoverContent.displayName = PopoverPrimitive.Content.displayName;
+
+export type PopoverProps = PopoverVariants & {
+  trigger: React.ReactNode;
+  children: React.ReactNode;
+  popoverRootProps?: React.ComponentPropsWithoutRef<typeof PopoveRoot>;
+  popoverTriggerProps?: React.ComponentPropsWithoutRef<typeof PopoverTrigger>;
+  popoverContentProps?: React.ComponentPropsWithoutRef<typeof PopoverContent>;

```

Comment:
I was thinking, maybe we can adopt the same implementation as radix and let the consumer do

Popover
  Popover.Trigger
  Popover.Content

This way, they can add props for trigger, content or popover without using specific props they have to relearn. Wdyt?

Reply:
sure! updated at b54aba4